lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -128,7 +128,7 @@ def build_checkout_request(amount, payment, options)
       end
 
       def add_payment(payment, options)
-        ntid = options.dig(:stored_credential, :network_transaction_id)
+        ntid = options.dig(:stored_credential, :network_transaction_id) if options[:stored_credential]
```

Comment:
There is not need of the `if` check because the `dig` method is being used

Reply:
you're right I will remove it, thanks